tools/stoc/makeTableAnalyse.xml

```diff
@@ -0,0 +1,71 @@
+<tool id="stoceps_maketablecarrer" name="Preprocess population data" version="@VERSION@">
+    <description>for evolution trend analyzes</description>
+    <macros>
+        <import>stoceps_macros.xml</import>
+    </macros>
+    <expand macro="reshape_requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        Rscript 
+         '$__tool_directory__/ExemakeTableAnalyseGalaxy.r' 
+         '$input'
+         '$__tool_directory__/FunctTrendSTOCGalaxy.r' 
+         '$output'
+    ]]>
+    </command>
+    <inputs>
+        <param name="input" type="data" format="tabular" label="Input file" help="Population count file, with location, date, species and abundance."/>
+    </inputs>
+    <outputs>
+        <data name="output" from_work_dir="Datatransformedforfiltering_trendanalysis.tabular" format="tabular"/>
+    </outputs>
+    <tests>
+        <test> 
+            <param name="input" value="data_FrenchBBS_squarre_dataSTOCallSp_France_trend_2001_2017_ALAARV-PARCAE_2001_2018.tabular"/>
+            <output name="output" file="Datatransformedforfiltering_trendanalysis.tabular"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+===============================
+STOC preprocess population data
+===============================
+
+**What it does**
+
+Reshape the data for the next steps of STOC analyzes by adding zero count to sites that are in the dataset, for species having no count data for those sites.
```

Comment:
@bgruening Yes definitely interesting! Thank you.

Reply:
Thank you very much. I have to look into details but yes it seems very interesting! And a lot of ecology data handling steps use this kind of functionality! Thanks!!!!

Reply:
this tool seems to be very powerfull... but I don't find a way to do what this step is doing: when a site don't have information about a species (line is missing) so we have to create a line with a 0 value (if I'm correct).

Reply:
hmm in fact no. This tool "just" reorganize the file. It take the third column classes and put each class as a new column header + fill the content with the related fourth column content

Reply:
hmmmm (re ;) ) . In fact this "table_compute" tool is amazing, and complex ;) so it appears to me this can be done through the inverse of melt function..

Reply:
So, Pivot seems to be the good operation to do, but need a unique identifier column... And here, we don't have this. I can compute one combining the 2 first columns but this will render the thing more complicated, and at least, I think of interest to propose 2 ways to do this step, using this STOC tool OR using table_compute after an additional step generating a unique ID column...  Maybe we can also work to add to table_compute tool the possibility, for Pivot functionality, to use combination of columns to have uniq ID...